src/main/java/org/sagebionetworks/web/client/SynapseJavascriptClient.java

```diff
@@ -0,0 +1,368 @@
+		String url = getRepoServiceUrl() + RESTRICTION_INFORMATION;
+		CallbackP<JSONObjectAdapter> constructCallback = new CallbackP<JSONObjectAdapter>() {
+			@Override
+			public void invoke(JSONObjectAdapter json) {
+				try {
+					callback.onSuccess(new RestrictionInformationResponse(json));
+				} catch (JSONObjectAdapterException e) {
+					callback.onFailure(e);
+				}
+			}
+		};
+		RestrictionInformationRequest request = new RestrictionInformationRequest();
+		request.setObjectId(subjectId);
+		request.setRestrictableObjectType(type);
+		try {
+			JSONObjectAdapter jsonAdapter = jsonObjectAdapter.createNew();
+			request.writeToJSONObject(jsonAdapter);
+			doPost(url, jsonAdapter.toJSONString(), wrapCallback(constructCallback, callback));
+		} catch (JSONObjectAdapterException e) {
+			callback.onFailure(e);
+		}
+	}
+	
+	public void getEntityChildren(EntityChildrenRequest request, final AsyncCallback<EntityChildrenResponse> callback) {
+		String url = getRepoServiceUrl() + ENTITY_URI_PATH + CHILDREN;
+		CallbackP<JSONObjectAdapter> constructCallback = new CallbackP<JSONObjectAdapter>() {
+			@Override
+			public void invoke(JSONObjectAdapter json) {
+				try {
+					callback.onSuccess(new EntityChildrenResponse(json));
+				} catch (JSONObjectAdapterException e) {
+					callback.onFailure(e);
+				}
+			}
+		};
+		try {
+			JSONObjectAdapter jsonAdapter = jsonObjectAdapter.createNew();
+			request.writeToJSONObject(jsonAdapter);
+			doPost(url, jsonAdapter.toJSONString(), wrapCallback(constructCallback, callback));
+		} catch (JSONObjectAdapterException e) {
+			callback.onFailure(e);
+		}
+	}
+	
+	public void getV2WikiPageAsV1(WikiPageKey key, AsyncCallback<WikiPage> callback) {
+		getVersionOfV2WikiPageAsV1(key, null, callback);
+	}
+	
+	public void getVersionOfV2WikiPageAsV1(final WikiPageKey key, final Long versionNumber, final AsyncCallback<WikiPage> callback) {
+		if (key.getWikiPageId() == null) {
+			// get the root wiki page id first
+			String url = getRepoServiceUrl() + "/" +
+					key.getOwnerObjectType().toLowerCase() + "/" + 
+					key.getOwnerObjectId() + WIKIKEY;
+			CallbackP<JSONObjectAdapter> constructCallback = new CallbackP<JSONObjectAdapter>() {
+				@Override
+				public void invoke(JSONObjectAdapter json) {
+					try {
+						String wikiPageId = new org.sagebionetworks.repo.model.dao.WikiPageKey(json).getWikiPageId();
+						key.setWikiPageId(wikiPageId);
+						getVersionOfV2WikiPageAsV1WithWikiPageId(key, versionNumber, callback);
+					} catch (JSONObjectAdapterException e) {
+						callback.onFailure(e);
+					}
+				}
+			};
+			doGet(url, wrapCallback(constructCallback, callback));
+		} else {
+			getVersionOfV2WikiPageAsV1WithWikiPageId(key, versionNumber, callback);
+		}
+	}
+	
+	private void getVersionOfV2WikiPageAsV1WithWikiPageId(WikiPageKey key, Long versionNumber, final AsyncCallback<WikiPage> callback) {
+		String url = getRepoServiceUrl() + "/" +
```

Comment:
Could this be converted to use v2 api?  (SWC could bundle into v1 object models if necessary)